aspnetcore/performance/memory.md

```diff
@@ -0,0 +1,408 @@
+---
+title: Memory management and patterns in ASP.NET Core
+author: rick-anderson
+description: Learn how memory is managed in ASP.NET Core and how the garbage collector (GC) works.
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/26/2019
+uid: performance/memory
+---
+
+# Memory management and garbage collection (GC) in ASP.NET Core
+
+By [Sébastien Ros](https://github.com/sebastienros) and [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Memory management is complex, even in a managed framework like .NET. Analyzing and understanding memory issues can be challenging. This article:
+
+* Was motivated by many *memory leak* and *GC not working* issues. Most of these issues were caused by not understanding how memory consumption works in .NET Core, or not understanding how it's measured.
+* Demonstrates problematic memory use, and suggests alternative approaches.
+
+## How garbage collection (GC) works in .NET Core
+
+The GC allocates heap segments where each segment is a contiguous range of memory. Objects placed in the heap are categorized into one of 3 generations: 0, 1, or 2. The generation determines the frequency the GC attempts to release memory on managed objects that are no longer referenced by the app. Lower numbered generations are GC'd more frequently.
+
+Objects are moved from one generation to another based on their lifetime. As objects live longer, they are moved into a higher generation. As mentioned previously, higher generations are GC'd less often. Short term lived objects always remain in generation 0. For example, objects that are referenced during the life of a web request are short lived. Application level singletons generally move to generation 1 and eventually 2.
+
+When an ASP.NET Core app starts, the GC:
+
+* Reserves some memory for the initial heap segments.
+* Commits a small portion of memory when the runtime is loaded.
+
+The preceding memory allocations are done for performance reasons. The performance benefit comes from heap segments in contiguous memory.
+
+**ASP.NET Core apps allocate a significant amount of memory at startup for better performance.**
```

Comment:
as a reader I'd want to know how this achieves better perf - are you caching something?

Reply:
@sebastienros can you answer this?

Reply:
Yea, I'm a bit confused by this statement as well.

Reply:
I was probably trying to simplify the fact that the Server GC allocates some memory per core (budget). Nothing to do with ASP.NET per se. With retrospect this statement is totally wrong.